pandas/_libs/tslibs/arraylike.pyx

```diff
@@ -0,0 +1,717 @@
+# -*- coding: utf-8 -*-
+
+cimport numpy as cnp
+from numpy cimport int64_t, ndarray, float64_t
+import numpy as np
```

Comment:
so some of this should be put in conversion. or is there a reason you are not doing that? i would like to shy away from creating a new module unless it has a clear purpose. can you move what is logical first.